### PR TITLE
CB-20112 Remove Jaeger dependencies and functionality

### PR DESCRIPTION
**Pull request by daszabo**
In preparation for FedRamp Jaeger should be removed as it is rarely used and presents some vulnerabilities.

### CODE REVIEW

File: cloud-gcp/build.gradle [hunk 1]
```diff
@@ -23,6 +23,7 @@ dependencies {
     implementation group: 'com.dyngr',                     name: 'polling',                                    version: dyngrPollingVersion
     implementation group: 'org.slf4j',                     name: 'slf4j-api',                                  version: slf4jApiVersion
     implementation group: 'org.apache.commons',            name: 'commons-lang3',                              version: apacheCommonsLangVersion
+    implementation group: 'org.jetbrains',                 name: 'annotations',                                version: '23.1.0'
```
Comment (lacikaaa) [inline cloud-gcp/build.gradle:26]: could you remove this?

File: cloud-template/build.gradle [hunk 2]
```diff
@@ -19,6 +19,7 @@ dependencies {
     implementation project(':cloud-reactor')
 
     implementation group: 'org.slf4j',                     name: 'slf4j-api',                      version: slf4jApiVersion
+    implementation group: 'org.jetbrains',                 name: 'annotations',                    version: '23.1.0'
```
Comment (lacikaaa) [inline cloud-template/build.gradle:22]: from everywhere

File: cloud-gcp/build.gradle (hunk 1 above)
Comment (daszabo) [reply cloud-gcp/build.gradle:26]: So should I remove the @NotNull annotations as well? eg. https://github.com/hortonworks/cloudbreak/blob/43c1b0ccc3d7e5cbf78de89bf6b42d7f26f4e810/common/src/main/java/com/sequenceiq/cloudbreak/logger/RestLoggerFilter.java#L78

Comment (lacikaaa) [reply cloud-gcp/build.gradle:26]: right, those intellij annotations shouldn't be there at all

Comment (lacikaaa) [reply cloud-gcp/build.gradle:26]: we might need a checkstyle rule to forbid intellij annotations

**Review by bergerdenes — CHANGES_REQUESTED** [review]

**Review by bergerdenes — APPROVED** [review]

**Pull request merged**